src/controllers/auth.controller.js

```diff
@@ -0,0 +1,203 @@
+'use strict';
+
+const { ApiError } = require('../exceptions/ApiError');
+const userService = require('../services/user.service');
+const jwtService = require('../services/jwt.service');
+const tokenService = require('../services/token.service');
+const { User } = require('../models/user.model');
+const bcrypt = require('bcrypt');
+const { randomUUID } = require('crypto');
+const { sendResetPassword } = require('../services/email.service');
+const refreshPasswordService
+= require('../services/refreshPasswordToken.service');
+
+function validateEmail(value) {
+  if (!value) {
+    return 'Email is required';
+  }
+
+  const emailPattern = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!emailPattern.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+function validatePassword(value) {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+}
+
+function validateName(value) {
+  if (!value) {
+    return 'Name is required';
+  }
+}
+
+async function register(req, res) {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    name: validateName(name),
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.email || errors.password || errors.name) {
```

Comment:
for the shortcut:
```suggestion
  if (Object.values(errors).length) {
```

Reply:
This `error` object contains 3 values in any case